cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1995,6 +1995,25 @@ func (p *BicepProvider) ensureParameters(
 			}
 		}
 
+		if param.IsAutoGen() {
+			// IsAutoGen already checked the azdMetadata exists
+			azdMetadata, _ := param.AzdMetadata()
```

Comment:
Just wondering -- maybe we don't need the helper method `IsAutoGen` at all? Easier to work with here IMO.